mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -229,6 +233,120 @@ impl TendermintCoin {
             .parse()
             .map_to_mm(|e| TendermintCoinRpcError::InvalidResponse(format!("balance is not u64, err {}", e)))
     }
+
+    #[allow(dead_code)]
+    fn gen_create_htlc_tx(
+        &self,
+        base_denom: Denom,
+        to: &AccountId,
+        amount: cosmrs::Decimal,
+        secret_hash: &[u8],
+        time_lock: u64,
+    ) -> MmResult<IrisHtlc, TxMarshalingErr> {
+        let timestamp = get_utc_timestamp() as u64;
+        let mut hash_lock_hash = vec![];
+        hash_lock_hash.extend_from_slice(secret_hash);
+        hash_lock_hash.extend_from_slice(&timestamp.to_be_bytes());
+
+        let amount = vec![Coin {
```

Comment:
Because id generation is depend on the `vec`, I wanted to keep it's type as it is even if we don't need it. So someday if we add extra coin in this vector, we don't need to touch id generation flow.